packages/ui/src/app/pages/WorkingGroups/WorkingGroupsOpenings.tsx

```diff
@@ -43,7 +48,23 @@ export const WorkingGroupsOpenings = () => {
       count: upcomingOpenings.length,
     },
   ]
-  const sideNeighborRef = useRef<HTMLDivElement>(null)
+
+  const displayOpenings = (type: 'CURRENT' | 'UPCOMING') => {
+    if ((type === 'CURRENT' && currentLoading) || (type === 'UPCOMING' && upcomingLoading)) {
+      return <Loading />
+    }
+
+    if (!(type === 'CURRENT' ? currentOpenings : upcomingOpenings).length) {
+      return <TextBig>No {type.toLowerCase()} openings found</TextBig>
+    }
+
+    return type === 'CURRENT' ? (
+      <OpeningsList openings={currentOpenings} />
+    ) : (
+      <UpcomingOpeningsList openings={upcomingOpenings} />
+    )
+  }
```

Comment:
I'm not sure if this is super easy to read. Maybe something like  

```tsx
  const displayOpenings = <T extends BaseOpening>(
    loading: boolean,
    openings: T[],
    Component: JSXElementConstructor<{ openings: T[] }>
  ) => {
    if (loading) {
      return <Loading />
    }

    if (openings.length) {
      return <TextBig>No openings found</TextBig>
    }

    return <Component openings={openings} />
  }
```

then:

```tsx
{activeTab === 'OPENINGS' && displayOpenings(upcomingLoading, upcomingOpenings, UpcomingOpeningsList)}
{activeTab === 'UPCOMING' && displayOpenings(currentLoading, currentOpenings, OpeningsList)}
```

Then you can also move this function out of the main render function scope, so fewer function creations as well.

Reply:
And it looks like component :upside\_down\_face: 

`<OpeningsTab isLodaing={upcomingLoading} openings={upcomingOpenings} component={UpcomingOpenings} />`

Reply:
Agree, looks just awesome!